server/com.dexels.navajo.compiler.tsl/src/com/dexels/navajo/compiler/tsl/internal/BundleQueueComponent.java

```diff
@@ -202,13 +226,10 @@ private void enqueueDependentScripts(String script, Set<String> history) {
 
         }
         for (String depScript : dependentScripts) {
-            if (history.contains(depScript)) {
-                logger.warn("Circular include dependency found! history: {} new: {}", history, depScript);
-                return;
-            }
```

Comment:
Recursion does still happen, it's just a bit more complex now (a calls b calls a calls b). Why remove the infinite loop protection?